outputs.tf

```diff
@@ -1,55 +1,54 @@
 output "specific_subnet_name" {
+  value       = azurerm_subnet.subnet[*].name
   description = "The name of the subnet."
-  value       = azurerm_subnet.subnet2.*.name
 }
 
 output "specific_subnet_id" {
-  description = "The subnet ID."
-  value       = azurerm_subnet.subnet2.*.id
+  value       = azurerm_subnet.subnet[*].id
+  description = "The name of the subnet."
 }
 
 output "specific_subnet_address_prefixes" {
   description = "The address prefixes for the subnet."
-  value       = azurerm_subnet.subnet2.*.address_prefixes
+  value       = azurerm_subnet.subnet[*].address_prefixes
 }
 
 output "default_subnet_name" {
+  value       = azurerm_subnet.subnet[*].name
   description = "The name of the subnet."
-  value       = azurerm_subnet.subnet.*.name
 }
 
 output "default_subnet_id" {
   description = "The subnet ID."
-  value       = azurerm_subnet.subnet.*.id
+  value       = azurerm_subnet.subnet[*].id
 }
 
 output "default_subnet_address_prefixes" {
+  value       = azurerm_subnet.subnet[*].address_prefixes
   description = "The address prefixes for the subnet."
-  value       = azurerm_subnet.subnet.*.address_prefixes
 }
 
 output "nat_gateway_id" {
+  value       = azurerm_nat_gateway.natgw[*].id
   description = "The ID of the NAT Gateway."
-  value       = azurerm_nat_gateway.natgw.*.id
 }
 
 output "public_ip_address" {
+  value       = azurerm_public_ip.pip[*].ip_address
```

Comment:
only 1 public ip is getting created, can we replace [*] with [0].

Reply:
Error: Invalid index
│ 
│   on ../../outputs.tf line 42, in output "public_ip_id":
│   42:   value       = azurerm_public_ip.pip[0].id
│     ├────────────────
│     │ azurerm_public_ip.pip is empty tuple
│ 
│ The given key does not identify an element in this collection value: the
│ collection has no elements.
╵